File: Factorial.py
#Iterativo
def factorial (n):
    fact = 1

    for i in range (n):
        fact = i * fact
        return fact

print(factorial(1))

#Recursivo
def factorial_recursivo(n):
    n=1
    if n == 1:
        return 1
    
    else:
        return factorial_recursivo(n-1) * n

File: Factorial.java
public class Factorial {

    public int factorial (int n){
        if (n < 0) {
            throw new IllegalArgumentException("n debe ser >= 0");
        }

        int fact = 1;

        for (int i = 1; i <= n; i++) {
            fact = i * fact;
        }

        return fact;
    }

    // Versión recursiva
    public int factorialRecursive(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("n debe ser >= 0");
        }

        if (n == 0) {
            return 1; // caso base
        }

        return n * factorialRecursive(n - 1); // llamada recursiva
    }

    // Método main de ejemplo
    public static void main(String[] args) {
        Factorial f = new Factorial();
        int n = 5;
        System.out.println("Iterativo " + n + "! = " + f.factorial(n));
        System.out.println("Recursivo " + n + "! = " + f.factorialRecursive(n));
    }
}